### PR TITLE
Fix coverage report testing on macOS

### LINKED ISSUE #43952: `make test KUBE_COVER=y` failed on Mac OS
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

**Kubernetes version** (use `kubectl version`):

**Environment**:
- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:
```
$ make test KUBE_COVER=y
Running tests for APIVersion: v1,apps/v1beta1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2alpha1,batch/v1,batch/v2alpha1,certificates.k8s.io/v1beta1,extensions/v1beta1,imagepolicy.k8s.io/v1alpha1,policy/v1beta1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,federation/v1beta1
+++ [0402 08:16:01] Saving coverage output in '/tmp/k8s_coverage/v1,apps/v1beta1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2alpha1,batch/v1,batch/v2alpha1,certificates.k8s.io/v1beta1,extensions/v1beta1,imagepolicy.k8s.io/v1alpha1,policy/v1beta1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,federation/v1beta1/20170402-081601'
hack/make-rules/test.sh: line 242: /bin/mkdir: Argument list too long
!!! [0402 08:16:01] Call tree:
!!! [0402 08:16:01]  1: hack/make-rules/test.sh:343 runTests(...)
make: *** [test] Error 1
```

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
Fixes coverage testing on macOS

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #43952

**Special notes for your reviewer**:
Uses SHA1 SUM of tests instead of full test string to generate directory, so that the directory doesn't exceed 255 chars.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/make-rules/test.sh [hunk 1]
```diff
@@ -186,7 +186,7 @@ junitFilenamePrefix() {
   # barely fits there and in coverage mode test names are
   # appended to generated file names, easily exceeding
   # 255 chars in length. So let's just sha1sum it.
```
Comment (gyliu513) [inline hack/make-rules/test.sh]: s/sha1sum/shasum with `-a 1` option.

One question is that I recalled this works well about a month ago, why it failed now?

Comment (cblecker) [inline hack/make-rules/test.sh]: I'm guessing that the length and number of APIs increased so that it now exceeds 255 characters

**Review by gyliu513 — COMMENTED** [review]
lgtm, thanks @cblecker

File: hack/make-rules/test.sh (hunk 1 above)
Comment (gyliu513) [inline hack/make-rules/test.sh]: make sense, thanks!

**Pull request merged**